### PR TITLE
Add capability to accept hex string in sign function

### PR DESCRIPTION
**Pull request by bsnowden3**
This PR adds the capability to add in a valid hexadecimal string in the schnorr sign function for ease of use.

### CODE REVIEW

File: src/check.js [hunk 1]
```diff
@@ -53,9 +53,17 @@ function checkNonceArr(nonces) {
 
 function checkPrivateKey(privateKey, idx) {
   const idxStr = (idx !== undefined ? '[' + idx + ']' : '');
-  if (!BigInteger.isBigInteger(privateKey)) {
-    throw new Error('privateKey' + idxStr + ' must be a BigInteger');
+  if (!BigInteger.isBigInteger(privateKey) && !(typeof privateKey == 'string')) {
+    throw new Error('privateKey' + idxStr + ' must be a BigInteger or string');
```
Comment (anders94) [inline src/check.js:57]: replace with ` or HEX string`?

Comment (bsnowden3) [reply src/check.js:57]: Good call

**Comment by guggero** [comment]
Thanks for submitting the PR. Could you perhaps elaborate a bit on why this should be added? And why only for the private key and not any of the other arguments?

**Comment by anders94** [comment]
This PR releases the requirement for code using `bip-schnorr` to also include `bigi` because use of the built-in `BigInteger` is becoming more common.

**Comment by bsnowden3** [comment]
Comment above refers to the use of other code using `bip-schnorr` being free of having to include `bigi` since built-in `BigInteger` is becoming more comm. Your point about the other arguments is valid. In my initial thought I only changed `privateKey` because it was `bigi` and `message` param is a buffer. Would changing the other two arguments be more useful and other  `bigi` instances be more likely to be accepted?

File: src/check.js [hunk 2]
```diff
@@ -53,9 +53,17 @@ function checkNonceArr(nonces) {
...
 function checkPrivateKey(privateKey, idx) {
   const idxStr = (idx !== undefined ? '[' + idx + ']' : '');
-  if (!BigInteger.isBigInteger(privateKey)) {
-    throw new Error('privateKey' + idxStr + ' must be a BigInteger');
+  if (!BigInteger.isBigInteger(privateKey) && !(typeof privateKey == 'string')) {
+    throw new Error('privateKey' + idxStr + ' must be a BigInteger or valid hex string');
   }
+
+  if (typeof(privateKey) == 'string') {
+    if ((privateKey.match(/[^a-f^A-F^0-9]+/))) {
```
Comment (guggero) [inline src/check.js:61]: nit: extra set of parenthesis.

File: src/schnorr.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ const zero = BigInteger.ZERO;
 function sign(privateKey, message, aux) {
   // https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#signing
   check.checkSignParams(privateKey, message);
+  privateKey = typeof (privateKey) == 'string' ? BigInteger.fromHex(privateKey) : privateKey;
```
Comment (guggero) [inline src/schnorr.js:18]: We should also update the pseudo API doc in the README to reflect this change.

**Review by guggero — CHANGES_REQUESTED** [review]
> Comment above refers to the use of other code using bip-schnorr being free of having to include bigi since built-in BigInteger is becoming more comm.

Ah, I see. Okay, that makes sense to me. JS isn't what I use daily so I didn't know about `BigInteger` now being built in.

Haven't looked at this code base in a while and thought that `BigInteger` was used as a parameter in more places. But it looks like that only applies to the MuSig code, which is more of a demo anyway.

So I think this change makes sense, just have a few nit pick comments :D

File: src/check.js (hunk 2 above)
Comment (bsnowden3) [reply src/check.js]: Ok awesome, changed

File: src/schnorr.js (hunk 3 above)
Comment (bsnowden3) [reply src/schnorr.js:18]: 👍

Comment (guggero) [reply src/schnorr.js:18]: Actually, I meant this: https://github.com/guggero/bip-schnorr/blob/master/README.md#schnorrsignprivatekey--biginteger-message--buffer--buffer
But that's okay, I can update that after merging.

**Review by guggero — APPROVED** [review]
LGTM, thanks!

**Pull request merged**